interpreter/function/builtin/digest_base64_decode.go

```diff
@@ -37,10 +41,63 @@ func Digest_base64_decode(ctx *context.Context, args ...value.Value) (value.Valu
 	}
 
 	input := value.Unwrap[*value.String](args[0])
-	dec, err := base64.StdEncoding.DecodeString(input.Value)
-	if err != nil {
-		return value.Null, err
+	removed := Digest_base64_decode_removeInvalidCharacters(input.Value)
+	dec, _ := base64.StdEncoding.DecodeString(removed)
+
+	return &value.String{Value: string(terminateNullByte(dec))}, nil
+}
+
+// Base64 decoding utility functions - they are also called by digest.base64url_decode and digest.base64_decode_nopad
+var nullByte = []byte{0}
+
+// Stop and return the point of Null-Byte found
+func terminateNullByte(decoded []byte) []byte {
+	if found := bytes.Index(decoded, nullByte); found != -1 {
+		return decoded[:found]
+	}
+	return decoded
+}
```

Comment:
This implementation works and this isn't important but the `strings.Cut` function would be a 1 line option for this that doesn't need any nesting.

```go
func terminateNullByte(decoded string) string {
    decoded, _, _ = strings.Cut(decoded, "\x00");
    return decoded;
}
```

Reply:
I will use `bytes.Cut` for it, thanks.